transaction/tests/AccessMode_unittest.cpp

```diff
@@ -0,0 +1,185 @@
+  EXPECT_FALSE(six_mode_.isCompatible(s_mode_));
+  EXPECT_FALSE(six_mode_.isCompatible(six_mode_));
+  EXPECT_FALSE(six_mode_.isCompatible(x_mode_));
+
+  EXPECT_TRUE(x_mode_.isCompatible(nl_mode_));
+  EXPECT_FALSE(x_mode_.isCompatible(is_mode_));
+  EXPECT_FALSE(x_mode_.isCompatible(ix_mode_));
+  EXPECT_FALSE(x_mode_.isCompatible(s_mode_));
+  EXPECT_FALSE(x_mode_.isCompatible(six_mode_));
+  EXPECT_FALSE(x_mode_.isCompatible(x_mode_));
+}
+
+TEST_F(AccessModeTest, ModeQueryChecks) {
+  EXPECT_FALSE(nl_mode_.isIntentionShareLock());
+  EXPECT_FALSE(nl_mode_.isIntentionExclusiveLock());
+  EXPECT_FALSE(nl_mode_.isShareLock());
+  EXPECT_FALSE(nl_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_FALSE(nl_mode_.isExclusiveLock());
+
+  EXPECT_TRUE(is_mode_.isIntentionShareLock());
+  EXPECT_FALSE(is_mode_.isIntentionExclusiveLock());
+  EXPECT_FALSE(is_mode_.isShareLock());
+  EXPECT_FALSE(is_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_FALSE(is_mode_.isExclusiveLock());
+
+  EXPECT_FALSE(ix_mode_.isIntentionShareLock());
+  EXPECT_TRUE(ix_mode_.isIntentionExclusiveLock());
+  EXPECT_FALSE(ix_mode_.isShareLock());
+  EXPECT_FALSE(ix_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_FALSE(ix_mode_.isExclusiveLock());
+
+  EXPECT_FALSE(s_mode_.isIntentionShareLock());
+  EXPECT_FALSE(s_mode_.isIntentionExclusiveLock());
+  EXPECT_TRUE(s_mode_.isShareLock());
+  EXPECT_FALSE(s_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_FALSE(s_mode_.isExclusiveLock());
+
+  EXPECT_FALSE(six_mode_.isIntentionShareLock());
+  EXPECT_FALSE(six_mode_.isIntentionExclusiveLock());
+  EXPECT_FALSE(six_mode_.isShareLock());
+  EXPECT_TRUE(six_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_FALSE(six_mode_.isExclusiveLock());
+
+  EXPECT_FALSE(x_mode_.isIntentionShareLock());
+  EXPECT_FALSE(x_mode_.isIntentionExclusiveLock());
+  EXPECT_FALSE(x_mode_.isShareLock());
+  EXPECT_FALSE(x_mode_.isShareAndIntentionExclusiveLock());
+  EXPECT_TRUE(x_mode_.isExclusiveLock());
+}
+
+TEST_F(AccessModeTest, Equality) {
+  const AccessMode nl_mode_expected = AccessMode(AccessModeType::kNoLock);
+  const AccessMode is_mode_expected = AccessMode(AccessModeType::kIsLock);
+  const AccessMode ix_mode_expected = AccessMode(AccessModeType::kIxLock);
+  const AccessMode s_mode_expected = AccessMode(AccessModeType::kSLock);
+  const AccessMode six_mode_expected = AccessMode(AccessModeType::kSixLock);
+  const AccessMode x_mode_expected = AccessMode(AccessModeType::kXLock);
+
+  const AccessMode nl_mode_checked = AccessMode(AccessModeType::kNoLock);
+  const AccessMode is_mode_checked = AccessMode(AccessModeType::kIsLock);
+  const AccessMode ix_mode_checked = AccessMode(AccessModeType::kIxLock);
+  const AccessMode s_mode_checked = AccessMode(AccessModeType::kSLock);
+  const AccessMode six_mode_checked = AccessMode(AccessModeType::kSixLock);
+  const AccessMode x_mode_checked = AccessMode(AccessModeType::kXLock);
```

Comment:
Remove all these `*_mode_expected` and `*_mode_checked`, and use data members in `AccessModeTest` directly, e.g., `nl_mode_`.

Thus it looks like:

```
  EXPECT_EQ(nl_mode_, nl_mode_);
  EXPECT_NE(nl_mode_, is_mode_);
```